CONTRIBUTING.md

```diff
@@ -158,3 +223,321 @@ including how to setup a local Kubernetes cluster, can be found in
+to see if an issue has already been created that addresses your concern.
+
+When creating your new issue, please ensure that you:
+
+* label the issue with the appropriate [`kind/<?>`](#kinds) label. The kind of
+  issue will determine the issue workflow.
+* label the issue with the appropriate [`area/<?>`](#area-assigment) label. The
+  area label will identify the area of interest an issue or PR belongs in and
+  will ensure the appropriate reviewers shepherd the issue or PR through to its
+  closure.
+* for issues, label the estimated size (if known). The size can be updated
+  during backlog grooming by the contributors. This estimate is used to guide
+  the number of features selected for a milestone.
+
+### Issue Triage
+
+Once an issue has been submitted, the CI (GitHub actions) or a human will
+automatically review the submitted issue or PR to ensure that it has all relevant
+information. If information is lacking or there is another problem with the
+submitted issue, an appropriate [`triage/<?>`](#triage) label will be
+applied.
+
+After an issue has been triaged, the maintainers can prioritize the issue with
+an appropriate [`priority/<?>`](#priority) label.
+
+Once an issue has been submitted, categorized, triaged, and prioritized it
+is marked as `ready-to-work`. A ready-to-work issue should have labels
+indicating assigned areas, prioritization, and should not have any remaining
+triage labels.
+
+### Issue and PR Kinds
+
+Use an kind label to describe the kind of issue or PR you are submitting. Valid
+kinds include:
+
+* [`kind/api-change`](#api-change) -- for api changes
+* [`kind/bug`](#bug) -- for filing a bug
+* [`kind/cleanup`](#cleanup) -- for code cleanup and organization
+* [`kind/deprecation`](#deprecation) -- for deprecating a feature
+* [`kind/design`](#design) -- for proposing a design or architectural change
+* [`kind/documentation`](#documentation) -- for updating documentation
+* [`kind/failing-test`](#failing-test) -- for reporting a failed test (may
+  create with automation in future)
+* [`kind/feature`](#feature) -- for proposing a feature
+* [`kind/support`](#support) -- to request support. You may also get support by
+  using our [Slack](https://kubernetes.slack.com/archives/CR2J23M0X) channel for
+  interactive help. If you have not setup the appropriate accounts, please
+  follow the instructions in [accounts setup](#accounts-setup).
+
+#### API Change
+
+A `kind/api-change` label categorizes an issue or PR as related to adding, removing,
+or otherwise changing an API.
+
+All API changes must be reviewed by MAINTAINERS in addition to the standard code
+review and approval workflow.
+
+To create an API change issue or PR:
+
+* label your issue or PR with `kind/api-change`
+* describe in the issue or PR body which API you are changing, making sure to include
+  * API endpoint and schema (endpoint, Version, APIGroup, etc.)
+  * Is this a breaking change?
```

Comment:
How is upgrade handled?

Reply:
Yes, upgrade should be documented here. The change would either handle upgrade automatically which means we need to have CI tests for that or we must include documentation in the release notes that describes the upgrade procedure. Adding the bullet point.